From a68e0428fdb0b34cdc7de673e2d0737fddf0d355 Mon Sep 17 00:00:00 2001
From: Aleksander Machniak <machniak@kolabsys.com>
Date: Fri, 4 Aug 2017 12:10:30 +0200
Subject: [PATCH 03/15] Bump SEQUENCE number on update (Outlook only).

It's been confirmed that any change of the event that has attendees specified
bumps SEQUENCE number of the event (we can see this in sent iTips).
---
 lib/kolab_sync_data_calendar.php | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/lib/kolab_sync_data_calendar.php b/lib/kolab_sync_data_calendar.php
index 946538a..8fab24d 100644
--- a/lib/kolab_sync_data_calendar.php
+++ b/lib/kolab_sync_data_calendar.php
@@ -332,6 +332,7 @@ class kolab_sync_data_calendar extends kolab_sync_data implements Syncroton_Data
         $foldername   = isset($event['_mailbox']) ? $event['_mailbox'] : $this->getFolderName($folderid);
         $config       = $this->getFolderConfig($foldername);
         $is_exception = $data instanceof Syncroton_Model_EventException;
+        $last_update  = $event['changed'];
 
         $event['allday'] = 0;
 
@@ -483,6 +484,16 @@ class kolab_sync_data_calendar extends kolab_sync_data implements Syncroton_Data
             $event['recurrence'] = $this->recurrence_to_kolab($data, $folderid, $timezone);
         }
 
+        // Bump SEQUENCE number on update (Outlook only).
+        // It's been confirmed that any change of the event that has attendees specified
+        // bumps SEQUENCE number of the event (we can see this in sent iTips).
+        if (!empty($entry) && !$is_exception && !empty($data->attendees)
+            && $data->dtStamp && $last_update && $data->dtStamp > $last_update
+            && stripos($this->device->devicetype, 'outlook') !== false
+        ) {
+            $event['sequence'] += 1;
+        }
+
         return $event;
     }
 
-- 
2.13.2

